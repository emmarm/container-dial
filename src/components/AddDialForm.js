import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../actions';
import getFavicon from '../utils/getFavicon';

class AddDialForm extends Component {
  state = {
    siteName: '',
    siteUrl: ''
  };

  onSiteNameChange = e => {
    const siteName = e.target.value;
    this.setState(() => ({ siteName }));
  };

  onSiteUrlChange = e => {
    const siteUrl = e.target.value;
    this.setState(() => ({ siteUrl }));
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { siteName, siteUrl } = this.state;
    const { container, addDial, handleHideAddDialModal } = this.props;
    const favicon = await getFavicon(siteUrl);
    const dial = {
      siteName,
      siteUrl,
      container: container.name,
      favicon
    };

    addDial(dial);
    handleHideAddDialModal(e);
  };

  render() {
    return (
      <form className="add-dial-modal__form">
        <div className="add-dial-modal__field">
          <label htmlFor="site-name" className="field__label">
            Site Name
          </label>
          <input
            id="site-name"
            className="field__input"
            type="text"
            placeholder="e.g. Facebook"
            onChange={this.onSiteNameChange}
            value={this.state.siteName}
          />
        </div>

        <div className="add-dial-modal__field">
          <label htmlFor="site-url" className="field__label">
            Site URL
          </label>
          <input
            id="site-url"
            className="field__input"
            type="text"
            placeholder="e.g. https://facebook.com"
            onChange={this.onSiteUrlChange}
            value={this.state.siteUrl}
          />
        </div>

        <div className="add-dial-modal__buttons">
          <button
            onClick={this.props.handleHideAddDialModal}
            className="add-dial-modal__button button--secondary"
          >
            Cancel
          </button>

          <button
            className="add-dial-modal__button button--primary"
            onClick={this.handleSubmit}
            style={{ backgroundColor: this.props.theme.primary }}
          >
            Add Dial
          </button>
        </div>
      </form>
    );
  }
}

AddDialForm.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  theme: PropTypes.objectOf(PropTypes.string)
};

const mapStateToProps = state => ({
  container: state.container
});

export default connect(
  mapStateToProps,
  actions
)(AddDialForm);
