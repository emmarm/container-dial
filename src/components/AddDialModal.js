import React from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';

const AddDialModal = ({ handleHideAddDialModal, container, isOpen, theme }) => (
  <Modal
    isOpen={isOpen}
    onRequestClose={handleHideAddDialModal}
    closeTimeoutMS={150}
    className="add-dial-modal"
  >
    <h2 className="add-dial-modal__title">
      New{' '}
      <span className="title__span" style={{ color: theme.primary }}>
        {container.name}
      </span>{' '}
      Dial
    </h2>

    <form className="add-dial-modal__form">
      <div className="add-dial-modal__field">
        <label htmlFor="site-name" className="field__label">
          Site Name
        </label>
        <input
          id="site-name"
          className="field__input"
          type="text"
          placeholder="e.g. Facebook"
        />
      </div>

      <div className="add-dial-modal__field">
        <label htmlFor="site-url" className="field__label">
          Site URL
        </label>
        <input
          id="site-url"
          className="field__input"
          type="text"
          placeholder="e.g. https://facebook.com"
        />
      </div>

      <div className="add-dial-modal__buttons">
        <button
          onClick={handleHideAddDialModal}
          className="add-dial-modal__button button--secondary"
        >
          Cancel
        </button>

        <button
          className="add-dial-modal__button button--primary"
          style={{ backgroundColor: theme.primary }}
        >
          Add Dial
        </button>
      </div>
    </form>
  </Modal>
);

AddDialModal.propTypes = {
  handleHideAddDialModal: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  container: PropTypes.objectOf(PropTypes.string)
};

export default AddDialModal;
