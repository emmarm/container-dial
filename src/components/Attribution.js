import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Text = styled('p')({
  color: 'white',
  fontSize: 14,
  fontStyle: 'italic',
  fontWeight: 100,
  marginRight: 40,
  textAlign: 'right'
});

const Link = styled('a')(
  {
    textDecoration: 'none'
  },
  ({ theme }) => ({
    color: theme.dark
  })
);

const Attribution = ({ image }) => {
  const { user } = !!image && image;
  return (
    <Text>
      Photo by{' '}
      <Link
        href={`${user &&
          user.links.html}?utm_source=container_dial&utm_medium=referral`}
      >
        {user && user.first_name} {user && user.last_name}
      </Link>{' '}
      on{' '}
      <Link href="https://unsplash.com/?utm_source=container_dial&utm_medium=referral">
        Unsplash
      </Link>
    </Text>
  );
};

Attribution.propTypes = {
  image: PropTypes.object
};

export default Attribution;
