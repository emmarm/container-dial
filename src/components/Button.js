import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Icon from './Icon';

const Btn = styled('button')(
  {
    alignItems: 'center',
    border: 'none',
    borderRadius: 4,
    boxShadow:
      'rgba(50, 50, 93, 0.1) 0px 7px 14px, rgba(0, 0, 0, 0.08) 0px 3px 6px',
    color: 'white',
    cursor: 'pointer',
    display: 'flex',
    fontSize: 20,
    fontWeight: 100,
    height: 50,
    justifyContent: 'center',
    padding: '0 25px',
    ':disabled': {
      backgroundColor: '#777',
      cursor: 'auto'
    }
  },
  ({ danger, narrow, padded, primary, theme }) => ({
    backgroundColor: primary ? theme.primary : danger ? '#FF3762' : '#bbb',
    padding: narrow && '0 15px',
    paddingLeft: padded && 30,
    ':hover': {
      backgroundColor: primary ? theme.light : danger ? '#FF5F81' : '#c8c8c8',
      ':disabled': {
        backgroundColor: '#777'
      }
    }
  })
);

const Button = ({
  className,
  danger,
  disabled,
  icon,
  narrow,
  onClick,
  primary,
  submit,
  text
}) => (
  <Btn
    className={className}
    danger={danger}
    disabled={disabled}
    onClick={onClick}
    narrow={narrow}
    padded={!!icon && !!text}
    primary={primary}
    type={submit ? 'submit' : 'button'}
  >
    {text}
    {icon && <Icon icon={icon} padded={!!text} />}
  </Btn>
);

Button.propTypes = {
  className: PropTypes.string,
  danger: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  submit: PropTypes.bool,
  text: PropTypes.string
};

export default Button;
