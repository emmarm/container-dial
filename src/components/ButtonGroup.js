import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

const Group = styled('div')({
  display: 'flex',
  justifyContent: 'space-around',
  margin: '25px'
});

const ButtonGroup = ({ children }) => <Group>{children}</Group>;

ButtonGroup.propTypes = {
  children: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
};

export default ButtonGroup;
