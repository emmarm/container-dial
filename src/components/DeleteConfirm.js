import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import * as actions from '../actions';

export const DeleteConfirm = ({
  dial,
  toggleShowDeleteConfirm,
  handleDelete,
  container,
  theme
}) => (
  <div>
    <p>
      Delete {dial.siteName} from {container.name}?
    </p>

    <button onClick={toggleShowDeleteConfirm}>Cancel</button>

    <button onClick={handleDelete}>Delete</button>
  </div>
);

DeleteConfirm.propTypes = {
  dial: PropTypes.objectOf(PropTypes.string),
  toggleShowDeleteConfirm: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  container: PropTypes.objectOf(PropTypes.string),
  theme: PropTypes.objectOf(PropTypes.string)
};

const mapStateToProps = state => ({
  container: state.container,
  theme: state.theme
});

export default connect(
  mapStateToProps,
  actions
)(DeleteConfirm);
