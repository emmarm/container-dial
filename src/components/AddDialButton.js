import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Button from './Button';

const AddButton = styled(Button)({
  background: 'rgba(255, 255, 255, 0.3)',
  borderRadius: 0,
  height: '100%',
  ':hover': {
    background: 'rgba(255, 255, 255, 0.5)'
  }
});

const AddDialButton = ({ handleShowDialModal }) => (
  <AddButton onClick={handleShowDialModal} text="Add New" icon="add" />
);

AddDialButton.propTypes = {
  handleShowDialModal: PropTypes.func.isRequired
};

export default AddDialButton;
