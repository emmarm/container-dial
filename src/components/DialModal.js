import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { css } from 'emotion';
import styled from 'react-emotion';

import * as actions from '../actions';
import DeleteConfirm from './DeleteConfirm';
import DialForm from './DialForm';

const modalClass = {
  base: css({
    alignItems: 'center',
    background: 'white',
    boxShadow:
      'rgba(50, 50, 93, 0.1) 0px 15px 35px, rgba(0, 0, 0, 0.07) 0px 5px 15px',
    display: 'flex',
    flexDirection: 'column',
    height: 'auto',
    margin: '15vh auto',
    minWidth: '420px',
    outline: 'none',
    padding: '30px',
    width: '50vw'
  }),
  beforeClose: css({
    display: 'none',
    transition: 'all 150ms ease-out'
  })
};

const modalOverlayClass = ({ theme }) =>
  css({
    background: `linear-gradient(to bottom, 
       ${theme.dark}33, ${theme.dark}33),
       linear-gradient(to bottom, rgba(0,0,0,0.8), rgba(0,0,0,0.8))`
  });

const DialTitle = styled('h2')({
  color: '#888',
  fontWeight: '100',
  marginBottom: '30px'
});

const Span = styled('span')(({ theme }) => ({
  color: theme.primary
}));

export class DialModal extends Component {
  state = {
    showDeleteConfirm: false
  };

  toggleShowDeleteConfirm = () => {
    this.setState(prevState => ({
      showDeleteConfirm: !prevState.showDeleteConfirm
    }));
  };

  handleClose = () => {
    this.setState(() => ({
      showDeleteConfirm: false
    }));
    this.props.handleHideDialModal();
  };

  handleDelete = () => {
    const { deleteDial, handleHideDialModal } = this.props;
    deleteDial(this.props.dial);
    this.toggleShowDeleteConfirm();
    handleHideDialModal();
  };

  render() {
    const { showDeleteConfirm } = this.state;
    const { isOpen, handleHideDialModal, dial, container } = this.props;
    return (
      <Modal
        className={modalClass}
        overlayClassName={modalOverlayClass}
        isOpen={isOpen}
        onRequestClose={this.handleClose}
        closeTimeoutMS={150}
      >
        {showDeleteConfirm ? (
          <DialTitle>Delete Dial</DialTitle>
        ) : dial ? (
          <DialTitle>Edit Dial</DialTitle>
        ) : (
          <DialTitle>
            New <Span>{container.name}</Span> Dial
          </DialTitle>
        )}

        {showDeleteConfirm ? (
          <DeleteConfirm
            dial={dial}
            handleDelete={this.handleDelete}
            toggleShowDeleteConfirm={this.toggleShowDeleteConfirm}
          />
        ) : (
          <DialForm
            dial={dial}
            handleHideDialModal={handleHideDialModal}
            toggleShowDeleteConfirm={this.toggleShowDeleteConfirm}
          />
        )}
      </Modal>
    );
  }
}

DialModal.propTypes = {
  handleHideDialModal: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  container: PropTypes.objectOf(PropTypes.string),
  dial: PropTypes.objectOf(PropTypes.string),
  deleteDial: PropTypes.func.isRequired
};

Modal.propTypes = {
  className: PropTypes.object,
  overlayClassName: PropTypes.func
};

const mapStateToProps = state => ({
  container: state.container
});

export default connect(
  mapStateToProps,
  actions
)(DialModal);
