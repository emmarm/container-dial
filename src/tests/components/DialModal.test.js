import React from 'react';
import { shallow } from 'enzyme';

import { DialModal } from '../../components/DialModal';

describe('DialModal component', () => {
  const props = {
    handleHideDialModal: jest.fn(),
    container: {
      name: 'Personal',
      color: 'orange'
    },
    isOpen: true,
    theme: {
      primary: '#333',
      light: '#fff',
      dark: '#000'
    },
    deleteDial: jest.fn()
  };
  const wrapper = shallow(<DialModal {...props} />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays correct title', () => {
    const title = wrapper
      .find('Styled(h2)')
      .dive()
      .find('h2')
      .text();
    const span = wrapper
      .find('Styled(span)')
      .dive()
      .find('span')
      .text();
    expect(span).toBe('Personal');
    expect(title).toBe('New <Styled(span) /> Dial');
  });

  it('renders DialForm', () => {
    expect(wrapper.find('Connect(DialForm)').length).toBe(1);
  });

  it('updates title on showDeleteConfirm', () => {
    wrapper.setState({ showDeleteConfirm: true });
    const title = wrapper
      .find('Styled(h2)')
      .dive()
      .find('h2')
      .text();
    expect(title).toBe('Delete Dial');
  });

  it('shows DeleteConfirm component on showDeleteConfirm', () => {
    wrapper.setState({ showDeleteConfirm: true });
    expect(wrapper.find('Connect(DialForm)').length).toBe(0);
  });

  describe('with dial prop', () => {
    const dial = {
      siteName: 'Goodreads',
      siteUrl: 'https://goodreads.com',
      container: 'Personal',
      favicon: 'image.jpg'
    };
    const wrapperWithDial = shallow(<DialModal {...props} dial={dial} />);

    it('updates title', () => {
      const title = wrapperWithDial
        .find('Styled(h2)')
        .dive()
        .find('h2')
        .text();
      expect(title).toBe('Edit Dial');
    });
  });
});
