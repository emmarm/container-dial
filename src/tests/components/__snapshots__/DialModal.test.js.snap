// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialModal component matches snapshot 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className={
    Object {
      "base": "css-5ws0ld",
      "beforeClose": "css-17m32ze",
    }
  }
  closeTimeoutMS={150}
  isOpen={true}
  onRequestClose={[Function]}
  overlayClassName={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <Styled(h2)>
    New 
    <Styled(span)>
      Personal
    </Styled(span)>
     Dial
  </Styled(h2)>
  <Connect(DialForm)
    handleHideDialModal={[Function]}
    toggleShowDeleteConfirm={[Function]}
  />
</Modal>
`;
